tests/test_dle.py

```diff
@@ -157,6 +158,27 @@ def test_cache_blocking_structure_subdims():
         tree[3].dim.root is y
     assert not tree[4].dim.is_Incr and tree[4].dim is zi and tree[4].dim.parent is z
 
+    # Non-local SubDimension -> nested blocking can works as expected
```

Comment:
this test is for nested parallelism, but you added it to `test_cache_blocking_...`. I think you should rather write a new test under `TestNestedParallelism`. This one is out of place

Reply:
also, does hierarchical blocking with nested parallelism over subdimensions work as expected? do we have one such test?

Reply:
Test moved. Nested+hierarchical+subdims seems ok, test added.